src/schemas/generator/index.ts

```diff
@@ -13,12 +13,29 @@ export function migrate(generator: z.infer<typeof AnyGeneratorSchema>) {
     case '0':
       return migrate(v0.migrate(generator))
     case '1.0':
+      migrateEmptyExtractionModeInCorrelation(generator)
       return generator
     default:
       return exhaustive(generator)
   }
 }
 
+// We added a new field `extractionMode` with a default value of `multiple` for the Correlation Rule
+// This rule was working in a `single` extraction mode previously so this migration checks that the value
+// is not defined and in that case we manually set the previous way of working
+function migrateEmptyExtractionModeInCorrelation(
```

Comment:
If this function is required, then:
1. It should be declared in one of the `v1` files, as it won't be needed in versions after that
2. it shouldn't mutate the generator object